src/core/lib/gprpp/map.h

```diff
@@ -209,6 +233,53 @@ class Map<Key, T, Compare>::iterator
   GrpcMap* map_;
 };
 
+template <class Key, class T, class Compare>
+class Map<Key, T, Compare>::const_iterator
+    : public std::iterator<std::input_iterator_tag, Pair<Key, T>, int32_t,
+                           Pair<Key, T>*, Pair<Key, T>&> {
+ public:
+  const_iterator(const const_iterator& iter)
+      : curr_(iter.curr_), map_(iter.map_) {}
+  bool operator==(const const_iterator& rhs) const {
+    return (curr_ == rhs.curr_);
+  }
+  bool operator!=(const const_iterator& rhs) const {
+    return (curr_ != rhs.curr_);
+  }
+
+  const_iterator& operator++() {
+    curr_ = map_->InOrderSuccessor(curr_);
+    return *this;
+  }
+
+  const_iterator operator++(int) {
```

Comment:
Maybe add a comment to distinguish the overloading operations, because the usage of `int` here is not obvious.

Reply:
Same here -- this was copied from the `iterator` class.